config.go

```diff
@@ -2196,11 +2211,29 @@ func configToFlatMap(cfg Config) (map[string]string, error) {
 			default:
 				continue
 			}
+
+			// If there's a hidden flag, it's deprecated.
+			if hidden == "true" && !field.IsZero() {
+				deprecated[key] = struct{}{}
+			}
 		}
 	}
 
 	// Turn the whole config struct into a flat map.
 	printConfig(reflect.ValueOf(cfg), "")
 
-	return result, nil
+	return result, deprecated, nil
+}
+
+// logWarningsForDeprecation logs a warning if a deprecated config option is
+// set.
+func logWarningsForDeprecation(cfg Config) {
+	_, configs, err := configToFlatMap(cfg)
```

Comment:
nit: call this `deprecated` instead of just `configs`?

Reply:
switched